### PR TITLE
Update Component Detection to .NET v8

### PR DESCRIPTION
**Pull request by cobya**
Update Component Detection to .NET v8, the latest LTS branch of .NET Core. See #771 for more information.

### CODE REVIEW

**Comment by cobya** [comment]
Dropping the SDK version back to 8.0.303 to avoid https://github.com/NuGet/Home/issues/13729

File: src/Microsoft.ComponentDetection.Detectors/npm/Contracts/PackageLockV1.cs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿#nullable enable
+#nullable enable
```
Comment (pauld-msft) [inline src/Microsoft.ComponentDetection.Detectors/npm/Contracts/PackageLockV1.cs:1]: Not super relevant to this pull request, but is there a reason that we don't have nullable configured at our project level with `<Nullable>enable</Nullable>`?

File: test/Microsoft.ComponentDetection.Common.Tests/BaseDetectionTelemetryRecordTests.cs [hunk 2]
```diff
@@ -73,7 +68,7 @@ public void SerializableProperties()
                 if (!serializableTypes.Contains(property.PropertyType) &&
                     Attribute.GetCustomAttribute(property.PropertyType, typeof(DataContractAttribute)) == null)
```
Comment (pauld-msft) [inline test/Microsoft.ComponentDetection.Common.Tests/BaseDetectionTelemetryRecordTests.cs:69]: dont need this second check anymore, doing this in the assertion below

**Review by pauld-msft — APPROVED** [review]

File: src/Microsoft.ComponentDetection.Detectors/npm/Contracts/PackageLockV1.cs (hunk 1 above)
Comment (cobya) [reply src/Microsoft.ComponentDetection.Detectors/npm/Contracts/PackageLockV1.cs:1]: We probably could - I'm not aware of any reasoning behind why we wouldn't allow it, but agree that this is out of scope for this PR

**Pull request merged**